src/main/java/net/mcreator/workspace/ReferencesFinder.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.workspace;
+
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.types.interfaces.IOtherModElementsDependent;
+import net.mcreator.element.types.interfaces.IResourcesDependent;
+import net.mcreator.element.types.interfaces.IXMLProvider;
+import net.mcreator.generator.LocalizationUtils;
+import net.mcreator.minecraft.DataListEntry;
+import net.mcreator.ui.workspace.resources.TextureType;
+import net.mcreator.util.FilenameUtilsPatched;
+import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.elements.SoundElement;
+import net.mcreator.workspace.resources.Model;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReferencesFinder {

```

Comment:
This object would benefit from its own test where workspace is created from eg. testworkspaceprovider and then some references are looked up

Reply:
Erm... I can try, but I've never wrote a test class before so expect me to at first do something wrong there.

Reply:
No worries, try and we will see how it goes / how much tests run longer due to it and then we decide if this test is needed or not depending on time cost to potential bugs found ratio

Reply:
Here it is.